subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/JavaGradlePluginPlugin.java

```diff
@@ -414,7 +414,7 @@ private AddOpensCommandLineArgumentProvider(Test test) {
 
         @Override
         public Iterable<String> asArguments() {
-            return test.getJavaVersion().isCompatibleWith(JavaVersion.VERSION_16)
+            return test.getJavaVersion().isCompatibleWith(JavaVersion.VERSION_1_9)
```

Comment:
Can we add an integ test for this to assert that we don't see any messages for Java 9+ and no failures for Java 16+?